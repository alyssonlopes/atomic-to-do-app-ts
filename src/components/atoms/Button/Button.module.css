.button {
  background-color: #0070f3;
  color: white;
  border: none;
  padding: 0.6rem 1.2rem;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.button:hover {
  background-color: #005bb5;
}
