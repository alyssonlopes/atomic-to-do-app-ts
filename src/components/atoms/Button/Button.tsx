import React from 'react';
import styles from './Button.module.css';

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  children: React.ReactNode;
};

const Button: React.FC<ButtonProps> = ({ children, ...props }) => {
  return (
    <button className={styles.button} {...props}>
      {children}
    </button>
  );
};

export default Button;
