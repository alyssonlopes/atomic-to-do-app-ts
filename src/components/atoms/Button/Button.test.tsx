import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('[Atom] Button Component', () => {
  it('renders button with text', () => {
    render(<Button>Click Me</Button>);
    const buttonElement = screen.getByText(/click me/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it('button click triggers onClick', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click Me</Button>);

    const buttonElement = screen.getByText(/click me/i);
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('renders button with different text', () => {
    render(<Button>Submit</Button>);
    const buttonElement = screen.getByText(/submit/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it('button click triggers different onClick handler', () => {
    const handleClick1 = jest.fn();
    const handleClick2 = jest.fn();
    render(<Button onClick={handleClick1}>Click Me</Button>);
    render(<Button onClick={handleClick2}>Submit</Button>);

    const buttonElement1 = screen.getByText(/click me/i);
    const buttonElement2 = screen.getByText(/submit/i);
    fireEvent.click(buttonElement1);
    fireEvent.click(buttonElement2);

    expect(handleClick1).toHaveBeenCalledTimes(1);
    expect(handleClick2).toHaveBeenCalledTimes(1);
  });

  it('button does not throw error without onClick handler', () => {
    render(<Button>Click Me</Button>);
    const buttonElement = screen.getByText(/click me/i);
    fireEvent.click(buttonElement);
    // No assertion needed, just ensuring no error is thrown
  });

  it('button does not trigger onClick when disabled', () => {
    const handleClick = jest.fn();
    render(
      <Button onClick={handleClick} disabled>
        Click Me
      </Button>
    );
    const buttonElement = screen.getByText(/click me/i);
    fireEvent.click(buttonElement);
    expect(handleClick).not.toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Button>Click Me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
