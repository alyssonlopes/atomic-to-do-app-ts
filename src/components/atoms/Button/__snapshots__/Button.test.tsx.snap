// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Atom] Button Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
